src/main/java/org/galatea/starter/domain/rpsy/IIexHistoricalPriceRpsy.java

```diff
@@ -1,27 +1,40 @@
 package org.galatea.starter.domain.rpsy;
 
 import java.util.List;
-import java.util.Optional;
 import org.galatea.starter.domain.IexHistoricalPrice;
 import org.springframework.cache.annotation.CacheEvict;
-import org.springframework.cache.annotation.Cacheable;
 import org.springframework.data.repository.CrudRepository;
 
 public interface IIexHistoricalPriceRpsy extends CrudRepository<IexHistoricalPrice, Long> {
 
   /**
-   * Retrieves all entities with the given symbol.
+   * Retrieves all entities with the given symbol and date.
    */
 
-  List<IexHistoricalPrice> findBySymbol(String symbol);
+  List<IexHistoricalPrice> findBySymbolAndDate(String symbol, String date);
 
-  @Override
-  @Cacheable(cacheNames = "historicalPrices", sync = true)
-  Optional<IexHistoricalPrice> findById(Long id);
+  /**
+   * Retrieves all entities with the given symbol and date in range.
+   */
+
+  List<IexHistoricalPrice> findBySymbolAndDateGreaterThanEqual(String symbol, String date);
+
+  /**
+   * Checks for the presence of an entity with the given data.
+   * @param symbol A string representing a symbol.
+   * @param date A string representing a date in YYYYMMDD format.
+   * @return True if the repository contains an entity with the given symbol and date; else, false.
+   */
+
+  boolean existsBySymbolAndDate(String symbol, String date);
```

Comment:
Are these parameters the same as the the findBySymbolAndDateGreaterThanEqual method above? If so, can you add the same Javadoc?

Reply:
Added Javadoc